app/models/workarea/payment/authorize/zipco.rb

```diff
@@ -46,6 +46,20 @@ def transaction_attrs
               capture: false
             }
           end
+
+          def authorize
+            request_id = SecureRandom.uuid
+            auth_response = response(request_id)
+            if Workarea::Zipco::RETRY_ERROR_STATUSES.include? auth_response.status
+              return response(request_id)
+            end
+
+            auth_response
+          end
+
+          def response(request_id)
```

Comment:
It'd probably be better to memoize something in case people decorate here.

```
  def response
    @response ||= begin
      # do all the stuff to get the response
    end
  end
```

Reply:
the response method has an argument so I'm not sure if memoization will work

Reply:
Yeah, you can memoize that too (that won't be called more than once when authorize runs).

Reply:
Response can be called more than once per request. 

The retry process is:
1 - call response.
2 - If its a 5xx error then call response again 
3 - just return whatever the second call is

Reply:
smdh 👌